src/main/java/utility/MarketTracker.java

```diff
@@ -119,9 +122,13 @@ public void updateStocks() {
             }
 
             // broadcast stock update to view
+            System.out.println("Broadcasting stock update...");
             ViewManager.Instance().broadcastEvent(new UpdateStockEvent(getStocks()));
 
-        } catch (RateLimitExceededException e) {
+            // notify observer of executionPrice update
+            System.out.println("Notifying observers...");
+            MarketObserver.Instance().onMarketUpdate();
+        } catch (RateLimitExceededException | IOException e) {
```

Comment:
I'm not really sure we can handle `IOException` the same way as `RateLimitExceededException`, as an `IOException` may capture a wider range of errors (such as connection error or corrupted data). It's better to handle it in a separate `catch` clause. For now we can just print out the error